spec/models/story_search_spec.rb

```diff
@@ -70,5 +84,28 @@
     it 'returns the foo labeled stories' do
       expect(StorySearch.new(project.stories, 'abc').search_labels).to eq([@story3, @story4])
     end
+
+    context 'search with labels' do
+      describe 'right labels' do
+        it 'one label' do
+          expect(StorySearch.new(project.stories, 'title: HELL').search).to eq([@story1, @story3])
+        end
+
+        subject(:search) { StorySearch.new(project.stories, 'title: HELL, labels: foo').search }
+        it 'two labels' do
+          expect(search).to eq([@story1])
+        end
+      end
+
+      describe 'wrong labels' do
+        it 'one label' do
```

Comment:
Can we make this more descriptive? I think  this can be done changing the text of the `describe` and the `it` and adding anothet `context` (we should do something like that to the prev one as well):

```rb
context 'when all labels used are valid' do
  context 'when the search matches a single result' do
    it 'returns only the result matched' do
      # ...
    end
  end
end
```